### PR TITLE
feat(markdown-docx): add conditional transformer - #397

### PR DESCRIPTION
**Pull request by K-Kumar-01**
Signed-off-by: k-kumar-01 <kushalkumargupta4@gmail.com>

Add the transformer `OOXML<->CiceroMark` for `conditional`.

### Changes

- transformation logic: OOXML<->CiceroMark
- rules: OPTIONAL_RULE
- tests

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [x] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `master` from `fork:branchname`

### CODE REVIEW

**Comment by K-Kumar-01** [comment]
@algomaster99 
Requesting your review.

File: packages/markdown-docx/src/ToCiceroMarkVisitor.js [hunk 1]
```diff
@@ -375,6 +403,15 @@ class ToCiceroMarkVisitor {
...
                 }
+                /**
+                 * for optional hasSome and whenSome stand for same as would be in the ciceromark.
+                 * for conditional, the following relations with optional can be made
+                 * optional    |     conditional
+                 * hasSome     |     isTrue
+                 * whenSome    |     whenTrue
+                 * whenFalse   |     whenFalse
+                 * The properties for conditional are not added therefore.
+                 */
```
Comment (algomaster99) [inline packages/markdown-docx/src/ToCiceroMarkVisitor.js:414]: Helps a lot. Thanks!

**Review by algomaster99 — APPROVED** [review]
Looks good to me as the changes look more or less similar to `Optional` node.

@dselman One question - what is the semantic difference between `Conditonal` and `Optional`? They look like they exist for the same purpose.

**Comment by algomaster99** [comment]
@K-Kumar-01 Ready for merge?

**Comment by K-Kumar-01** [comment]
@algomaster99 
Yes this one can be merged.
Didn't have many changes except the `keys`.
I provided an explanation regarding that in the comments itself.

**Pull request merged**